hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/cluster/strategy/ClusteringExecutionStrategy.java

```diff
@@ -18,25 +18,27 @@
 
 package org.apache.hudi.table.action.cluster.strategy;
 
-import org.apache.avro.Schema;
+import org.apache.hudi.avro.model.HoodieClusteringPlan;
 import org.apache.hudi.common.engine.HoodieEngineContext;
 import org.apache.hudi.common.model.HoodieRecordPayload;
 import org.apache.hudi.config.HoodieWriteConfig;
 import org.apache.hudi.table.HoodieTable;
+import org.apache.hudi.table.action.HoodieWriteMetadata;
+
+import org.apache.avro.Schema;
 import org.apache.log4j.LogManager;
 import org.apache.log4j.Logger;
 
 import java.io.Serializable;
-import java.util.Map;
 
 /**
  * Pluggable implementation for writing data into new file groups based on ClusteringPlan.
  */
-public abstract class ClusteringExecutionStrategy<T extends HoodieRecordPayload,I,K,O> implements Serializable {
+public abstract class ClusteringExecutionStrategy<T extends HoodieRecordPayload, I, K, O> implements Serializable {
```

Comment:
@satishkotha One change in this class is that I have moved the `transform()` method to corresponding strategy classes. There is some duplicate code but due to virtual key changes I had to do that. Otherwise, we need significant refactoring i.e. move the `HoodieSparkKeyGeneratorFactory`  and several key generators to hudi-client-common instead of hudi-spark-client.